Document the Unicode mode for Windows

alainfrisch opened the pull request:
This should of course be cherry-picked to 4.06.

(On a related note, a decision needs to be made of the default mode.   Currently, it is "Unicode".)

alainfrisch committed:
Document the Unicode mode for Windows.

alainfrisch committed:
Typo.

alainfrisch committed:
Typo.

nojb commented on README.win32.adoc:
```diff
@@ -339,6 +339,50 @@ compiling `world`, you must compile `flexdll`, i.e.:
+to be encoded using the current 8-bit code page of the system.  Some
+Unicode filenames could thus not be represented.  Since version 4.06,
+OCaml adds to this legacy mode a new "Unicode" mode, where filenames
+are interpreted as UTF-8 encoded strings.  In addition to filenames,
```
Would delete `interpreted as`.

nojb commented on README.win32.adoc:
```diff
@@ -339,6 +339,50 @@ compiling `world`, you must compile `flexdll`, i.e.:
+Unicode filenames could thus not be represented.  Since version 4.06,
+OCaml adds to this legacy mode a new "Unicode" mode, where filenames
+are interpreted as UTF-8 encoded strings.  In addition to filenames,
+this applies to environment variables and command-line arguments as well.
```
Would delete `as well` (it is redundant with `In addition`).

nojb commented on README.win32.adoc:
```diff
@@ -339,6 +339,50 @@ compiling `world`, you must compile `flexdll`, i.e.:
+   page.  This fallback works well in practice, since the chances of
+   non-ASCII string encoded in the a 8-bit code page to be a valid
+   UTF-8 string are tiny.  This means that filenames extracted
+   obtained e.g. from a 8-bit UI or database layer would continue to
```
`obtained e.g. from` -> `from e.g.` ?

nojb commented on README.win32.adoc:
```diff
@@ -339,6 +339,50 @@ compiling `world`, you must compile `flexdll`, i.e.:
+   UTF-8 string are tiny.  This means that filenames extracted
+   obtained e.g. from a 8-bit UI or database layer would continue to
+   work fine.  Application written for the legacy mode or older
+   versions of OCaml might still be broken if strings returned by
```
`be broken` -> `break` ?

alainfrisch committed:
Apply suggested changes.

alainfrisch commented:
@nojb Thanks for the suggestions.  Do you think we should also document the correct way to write C bindings for functions on "os strings"?  (Certainly not in the same file, probably either in the manual or in one of the public .h file.)

gasche commented:
Yes, I think the changes to the C API "good practices" should absolutely be documented after #1363 converges.

damiendoligez commented on README.win32.adoc:
```diff
@@ -339,6 +339,50 @@ compiling `world`, you must compile `flexdll`, i.e.:
+   page.  This fallback works well in practice, since the chances of
+   non-ASCII string encoded in the a 8-bit code page to be a valid
+   UTF-8 string are tiny.  This means that filenames extracted
+   obtained from e.g. a 8-bit UI or database layer would continue to
```
"extracted obtained" you should choose one of the two verbs

alainfrisch replied:
Thanks, fixed.

nojb commented:
I think we should also explain that successful use the native-code compiler with Unicode pathnames requires a "bootstrapped" flexlink: a flexlink that has been compiled using the new Unicode-enabled runtime.

alainfrisch committed:
Typo.

alainfrisch committed:
Instructions related to FlexDLL.

alainfrisch commented:
> I think we should also explain that successful use the native-code compiler with Unicode pathnames requires a "bootstrapped" flexlink: a flexlink that has been compiled using the new Unicode-enabled runtime.

Ok, added a sentence saying so.  Also added some words about the the FlexDLL vs Visual Studio versions.

xavierleroy commented:
(Putting my editor hat on:)  the reviewers's comments were adequately addressed, hence I am happy to accept your paper^H^H^H^H^H^H^H^H^H merge this PR in trunk and 4.06.

xavierleroy commented:
Cherry-picked as [4.06 12c370b]
